### DISCUSSION TITLE
Why dependencies are not installed automatically

### DISCUSSION THREAD

**Discussion by sunnysnowhi** [microsoft/vscode-cpptools#8578] [Q&A] [answered]
I want to install offline vscode-cpptools package 1.5.1 in Ubuntu 18.04
I download the package cpptools-linux.vsix from  (https://github.com/microsoft/vscode-cpptools/releases?page=2) 
And then I install this by command  (not root):
/home/my-user/VSCode-linux-x64/bin/code --install-extension  cpptools-linux.vsix
But when I checking the installed files in /home/my-user/,vscode/extensions/ms-vscode.cpptools-1.5.1,  no dependencies was found, such as LLVM folder

Then I add --user-data-dir to install the extension:
/home/my-user/VSCode-linux-x64/bin/code --install-extension  cpptools-linux.vsix --user-data-dir /home/my-user/.vscode-server/user-data
I found the dependencies were installed!

Why --user-data-dir can make it install dependencies automatically?   Since I use non-root user to install extension, I think it is unnessary to add  --user-data-dir.  But how can I install dependencies automatically without --user-data-dir?

Thank you.

**Comment by Colengms** [comment, marked as answer]
Hi @sunnysnowhi .  The 'offline' (platform-specific) VSIX files available on the release page will include all dependencies.  The 'online' VSIX or the one you get when installing via the Marketplace) will download dependencies when first activated.  I don't believe the `--user-data-dir` is directly related to this behavior.

A VSIX file is actually a ZIP file.  I've confirmed that the `cpptools-linux.vsix` for 1.5.1 does contain the dependencies.  If that VSIX was extracted, those files should be present.  Perhaps you had previously installed the online version of the VSIX, and were looking at those contents?

**Reply by sunnysnowhi** [reply]
yes, I installed too much times and messed up the installation package
